Content.Server/Polymorph/Systems/PolymorphSystem.cs

```diff
@@ -260,9 +260,11 @@ private void OnDestruction(Entity<PolymorphedEntityComponent> ent, ref Destructi
             _mindSystem.TransferTo(mindId, child, mind: mind);
 
         //Ensures a map to banish the entity to
-        EnsurePausedMap();
-        if (PausedMap != null)
-            _transform.SetParent(uid, targetTransformComp, PausedMap.Value);
+        if (PausedMapStorage.EnsurePausedMap())
+        {
+            PausedMapStorage.BeforeEnter(uid, child);
+            _transform.SetParent(uid, targetTransformComp, PausedMapStorage.PausedMap.Value);
```

Comment:
why does this need to be two calls and you have to remember to call `BeforeEnter` and `AfterExit`, shouldnt the methods on `PausedMapStorage` just be like `InsertIntoPausedStorage` and `RemoveFromPausedStorage`